utils.go

```diff
@@ -77,14 +77,11 @@ func Itob(integer int) []byte {
 
 // generate a random secret of given length
 func RandomSecret(length int) string {
-	rand.Seed(time.Now().UnixNano())
-	letterRunes := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ234567")
-
-	bytes := make([]rune, length)
-
-	for i := range bytes {
-		bytes[i] = letterRunes[rand.Intn(len(letterRunes))]
+	s := make([]byte, length*10)
+	_, err := rand.Read(s)
+	if err != nil {
+		log.Println("Unable to generate secret; insufficient entropy?",err)
+		return ""
 	}
-
-	return string(bytes)
+	return hex.EncodeToString(s[:length])
```

Comment:
random secret should be base32 format according to [OTP.secret](https://github.com/xlzd/gotp/blob/c8557ba2c11930b3069b31bfd80f38cc774cc68a/otp.go#L19).

Reply:
16 bytes with a constrained alphabet (36 possibilities) isn't a large enough search space to be secure.

So, it should really be at least 32 bytes -- and, really, going beyond that would be a good idea (that's what most other OTP libs that I've reviewed use). I was not able to find any limitations in the OTP specs themselves, so you should be able to go to 64 as well.

Here's some docs.

https://datatracker.ietf.org/doc/html/rfc4226#appendix-A.4.1
https://datatracker.ietf.org/doc/html/rfc6238
https://datatracker.ietf.org/doc/html/rfc4086#page-34
github.com/pquerna/otp/totp

Reply:
What if you just hash the result with SHA256?

Reply:
from https://github.com/pquerna/otp/blob/60112ee2a95553a491c22956619ef30260ec93e8/totp/totp.go#L150

```go
var b32NoPadding = base32.StdEncoding.WithPadding(base32.NoPadding)
b32NoPadding(EncodeToString(secret))
```

Still needs to be 32 bytes though.

Reply:
Trying to figure out how to edit this PR in the web UI without actually going to the trouble of cloning it locally haha